hyperSpec/vignettes/hyperSpec.Rmd

```diff
@@ -1828,124 +1475,267 @@ The matrix formulation puts the $n$ spectra into the rows of $A$ and $c$, while
 
-\mFun{`\%*\%`}
-If we decide that only the first 10 components actually carry spectroscopic information, we can reconstruct spectra with better signal to noise ratio:
+\mFun{`\%*\%`}
+If we decide that only the first 10 components actually carry spectroscopic information, we can reconstruct spectra with better signal to noise ratio:
+
+```{r pca-smooth}
+smoothed <- scores[, , 1:10] %*% loadings[1:10]
+```
+Keep in mind, though, that we cannot be sure how much *useful* information was discarded with the higher components.
+This kind of noise reduction may influence further modeling of the data. Mathematically speaking, the rank of the new `r nrow (chondro)` $×$ `r nwl (chondro)` spectra matrix is only 10.
+
+
+## Hierarchical Cluster Analysis {#HCA}
+
+\mFun{`[[]]`{.r}}
+Some `R` functions expect their input data in a matrix, including those related to HCA. Conveniently, the `[[]]`{.r} extraction function returns a matrix.
+
+```{r hca}
+dist <- pearson.dist(chondro[[]])
+```
+
+Again, many such functions coerce the data to a matrix automatically, so the `hyperSpec` object can be handed over without the explicit conversion above (as we saw for PCA):
+
+```{r hca-asmatrix}
+dist <- pearson.dist(chondro)
+dendrogram <- hclust(dist, method = "ward.D")
+```
+
+```{r include = FALSE}
+CAPTION <- "The results of the cluster analysis: the dendrogram."
+```
+
+```{r dend, fig.cap = CAPTION, sq.fig = TRUE}
+plot(dendrogram, labels = FALSE, hang = -1)
+```
+
+In order to plot a cluster map, the cluster membership needs to be calculated from the dendrogram.
+
+First, cut the dendrogram so that three clusters result:
+
+```{r dendcut}
+chondro$clusters <- as.factor(cutree(dendrogram, k = 3))
+```
+As the cluster membership was stored as factor, the levels can be meaningful names, which are displayed in the color legend.
+
+```{r clustname}
+levels(chondro$clusters) <- c("matrix", "lacuna", "cell")
+```
+
+Then the result may be plotted (Figure \@ref(fig:clust-map)):
+
+```{r include = FALSE}
+CAPTION <- "The results of the cluster analysis: the map of the 3 clusters.  "
+```
+
+```{r clust-map, fig.cap = CAPTION, sq.fig = TRUE}
+plotmap(chondro, clusters ~ x * y, col.regions = cluster.cols)
+```
+
+
+## Calculating group-wise Sum Characteristics, e. g. Cluster Mean Spectra
+
+\mFun{`aggregate()`{.r}}
+Function `aggregate()`{.r} applies the function given in `FUN`{.r} to each of the groups of spectra specified in `by`{.r}.
+
+So we may plot the cluster mean spectra:
+
+
+
+```{r include = FALSE}
+CAPTION <- "The results of the cluster analysis: the the mean spectra.  "
+```
+
+```{r clustmean, fig.cap = CAPTION}
+means <- aggregate(chondro, by = chondro$clusters, mean_pm_sd)
+plot(means, col = cluster.cols, stacked = ".aggregate", fill = ".aggregate")
+```
+
+# Combining and Splitting `hyperSpec` Objects {#sec:combine}
+
+## Binding Objects together {#sec:bind}
```

Comment:
Shouldn't it be in title case?

There are several more headings, that should be fixed into title case.

Reply:
I didn't list them all.

Reply:
Yes, they should.  Thanks for taking a careful look with fresh eyes.